test/test_support.ts

```diff
@@ -56,9 +57,10 @@ function transformSource(src: string): string {
   };
 
   var program = ts.createProgram(['main.ts'], OPTIONS, host);
-  if (program.getSyntacticDiagnostics().length) {
```

Comment:
This was only checking "SyntacticDiagnostics", while the latter code checked "PreEmitDiagnostics".  The latter is a superset of the former.